lib/taste_tester/commands.rb

```diff
@@ -210,5 +210,166 @@ def self.upload
+      # We want to compare changes in the current directory (working set) with
+      # the "most recent" commit in the VCS. For SVN, this will be the latest
+      # commit on the checked out repository (i.e. 'trunk'). Git/Hg may have
+      # different tags or labels assigned to the master branch, (i.e. 'master',
+      # 'stable', etc.) and should be configured if different than the default.
+      start_ref = case repo
+                  when BetweenMeals::Repo::Svn
+                    repo.latest_revision
+                  when BetweenMeals::Repo::Git
+                    TasteTester::Config.vcs_start_ref_git
+                  when BetweenMeals::Repo::Hg
+                    TasteTester::Config.vcs_start_ref_hg
+                  end
+      end_ref = TasteTester::Config.vcs_end_ref
+
+      changeset = BetweenMeals::Changeset.new(
+        logger,
+        repo,
+        start_ref,
+        end_ref,
+        {
+          :cookbook_dirs =>
+            TasteTester::Config.relative_cookbook_dirs,
+          :role_dir =>
+            TasteTester::Config.relative_role_dir,
+          :databag_dir =>
+            TasteTester::Config.relative_databag_dir,
+        },
+        @track_symlinks,
+      )
+
+      return changeset
+    end
+
+    def self._find_roles(changes)
+      if TasteTester::Config.relative_cookbook_dirs.length > 1
+        logger.error('Knife deps does not support multiple cookbook paths.')
+        logger.error('Flatten the repo or override the resolve_deps function.')
+        exit(1)
+      end
+
+      cookbooks = Set.new(changes.cookbooks)
+      roles = Set.new(changes.roles)
+      databags = Set.new(changes.databags)
+
+      if cookbooks.empty? && roles.empty?
+        logger.warn('No cookbooks or roles have been modified.')
+        return Set.new
+      end
+
+      unless cookbooks.empty?
+        logger.info('Modified Cookbooks:')
+        cookbooks.each { |cb| logger.info("\t#{cb}") }
+      end
+      unless roles.empty?
+        logger.info('Modified Roles:')
+        roles.each { |r| logger.info("\t#{r}") }
+      end
+      unless databags.empty?
+        logger.info('Modified Databags:')
+        databags.each { |db| logger.info("\t#{db}") }
+      end
+
+      # Use Knife to list the dependecies for each role in the roles directory.
+      # This creates a recursive tree structure that is then searched for
+      # instances of modified cookbooks. This can be slow since it must read
+      # every line of the Knife output, then search all roles for dependencies.
+      # If you have a custom way to calculate these reverse dependencies, this
+      # is the part you would replace.
+      logger.info('Finding dependencies (this may take a minute or two)...')
+      knife = Mixlib::ShellOut.new(
+        "knife deps /#{TasteTester::Config.role_dir}/*.rb" +
+        " --config #{TasteTester::Config.knife_config}" +
+        " --chef-repo-path #{TasteTester::Config.absolute_base_dir}" +
```

Comment:
we should file a bug upstream to knife - it's common to have multiple cookbook dirs, and `knife deps` should support this. Can you also do that please?